### ISSUE TITLE
The Claim.Issuer value is lost during claims building process.

### ISSUE THREAD

**Issue by Crimp** [Azure/azure-signalr#1507]
### Description
The Claim has an [Issuer](https://docs.microsoft.com/en-us/dotnet/api/system.security.claims.claim.issuer?view=net-6.0) parameter, which can be passed when you create a custom claim or from Google (like your e-mail address) and then the original issuer would be Google. That data can be processed by a custom logic.
When you add Azure SignalR connection to the application, you cannot check that value anymore, becouse it's replaced by defoult one "Local Authority", and can not obtain that value from the Claim.OriginalIssuer.

I looked at the
 [Microsoft.Azure.SignalR.ClaimsUtility.BuildJwtClaims](https://github.com/Azure/azure-signalr/blob/5ad081f12eafb2c84659790e2e7b6b0eb7e79645/src/Microsoft.Azure.SignalR.Common/Utilities/ClaimsUtility.cs#L119)
and 
 [Microsoft.Azure.SignalR.Protocols.dll!Microsoft.Azure.SignalR.Protocol.ServiceProtocol.ReadClaims](https://github.com/Azure/azure-signalr/blob/848bd685cafe00d4b05e418e9c06557903ce3618/src/Microsoft.Azure.SignalR.Protocols/ServiceProtocol.cs#L1144)

There are only claim.Type and claim.Value are used. This made me think that maybe this is the way it was intended?

### To Reproduce
We are using Blazor application.

The custom middleware can call the code below to login a user using some token processing logic:
```C#
IList<Claim> claims = new List<Claim>();
claims.Add(new Claim(ClaimTypes.NameIdentifier, "some user ID", ClaimValueTypes.String, "CustomIssuer"));
claims.Add(new Claim(ClaimTypes.Name, "TestName", ClaimValueTypes.String, "CustomIssuer"));
var user = new ClaimsPrincipal(new ClaimsIdentity(claims, SecurityDefaults.PasswordAuthentication));
await context.SignInAsync(user);
```

Later the application won't be able to obtain the Issuer from claims. The logic won't work:
```C#
if(((ClaimsPrincipal)user).HasClaim(c => c.Type == ClaimTypes.NameIdentifier && c.Issuer == "CustomIssuer") {
   ...
}
```

The same behaviour using Azure AD or any OAuth authentication, the Issuer will be ClaimsIdentity.DefaultIssuer anyway.

When ClaimsPrincipal has several ClaimsIdentity it also can have several claims with the same Claim.Type and it would be good to understand who is who. Of course you can read them from each ClaimsIdentity separetly. Just wondering is it by design and we should forget about Issuer?

### Exceptions (if any)
No exceptions, just missed my Issuer value.

### Further technical details
- I am using "Microsoft.Azure.SignalR" Version="1.4.0"
- net5.0

**Comment by terencefan** [comment]
Acknowledged.

You're right, `issuer` and `original issuer` won't be passed to server SDK since we didn't put these properties into the `OpenConnectionMessage`, which is used to generate connection context.

I'm not sure if it was intended, will discuss it with other team members later.

**Comment by LaurensAdema** [comment]
Was this ever discussed? Can they be added? We need both issuer and original issuer for our validation. Thanks in advance!

**Comment by LaurensAdema** [comment]
@terencefan any update on this?

**Comment by blarsern** [comment]
I don't get it, the issue was created back in 2021?
And still nothing has been done..

So what should we do, should i allow all claims by LOCAL AUTHORITY?
That seems like a huge security issue?

We are very strict and always check that the claim is from our Issuer, so all our logic breaks when in a signalr hub method.

Seems like a minor fix to me. And should have been fixed back in 2021.
What are the plans to fix this ? What can be more important than this?

@terencefan

**Comment by terencefan** [comment]
I linked a PR to this issue, but not sure if it will address the issue.

In this PR I applied the customer provided-issuer (asrs.u.iss) to all customer claims as their original issuer.

But it doesn't make sense in my opinion.

It is not possible for us to preserve original issuer for every claim.

According to [RFC7519](https://datatracker.ietf.org/doc/html/rfc7519), the jwt token contains only `type` and `name` field in its payload. 
This means when we create a token and return it in the negotiation response, the `issuer` and `orignal issuer` are lost, making it impossible to restore them from the token.

Claims in our case are only for transmitting data between the server, the client, ASRS, and back to the server.

Every security check, including issuer and original issuer checks, should be completed in your negotiate handler, not in the `OnConnected` event of a client connection.
